django/thunderstore/frontend/api/experimental/tests/test_community_package_list.py

```diff
@@ -0,0 +1,418 @@
+    listing2 = PackageListingFactory(
+        community_=listing1.community, review_status=PackageListingReviewStatus.approved
+    )
+    PackageListingFactory(
+        community_=listing1.community, review_status=PackageListingReviewStatus.rejected
+    )
+
+    data = __query_api(api_client, listing1.community.identifier)
+
+    __assert_packages_by_listings(data, [listing2, listing1])
+
+
+@pytest.mark.django_db
+def test_only_approved_packages_are_returned_when_approval_is_required(
+    api_client: APIClient,
+) -> None:
+    listing1 = PackageListingFactory(
+        review_status=PackageListingReviewStatus.unreviewed
+    )
+    listing1.community.require_package_listing_approval = True
+    listing1.community.save()
+    listing2 = PackageListingFactory(
+        community_=listing1.community, review_status=PackageListingReviewStatus.approved
+    )
+    PackageListingFactory(
+        community_=listing1.community, review_status=PackageListingReviewStatus.rejected
+    )
+
+    data = __query_api(api_client, listing1.community.identifier)
+
+    __assert_packages_by_listings(data, listing2)
+
+
+@pytest.mark.django_db
+def test_deprecated_packages_are_returned_only_when_requested(
+    api_client: APIClient,
+) -> None:
+    active = PackageListingFactory()
+    PackageListingFactory(
+        community_=active.community, package_kwargs={"is_deprecated": True}
+    )
+
+    data = __query_api(api_client, active.community.identifier)
+
+    __assert_packages_by_listings(data, active)
+
+    data = __query_api(api_client, active.community.identifier, "deprecated")
+
+    assert len(data["packages"]) == 2
+
+
+@pytest.mark.django_db
+def test_nsfw_packages_are_returned_only_when_requested(api_client: APIClient) -> None:
+    sfw = PackageListingFactory()
+    PackageListingFactory(community_=sfw.community, has_nsfw_content=True)
+
+    data = __query_api(api_client, sfw.community.identifier)
+
+    __assert_packages_by_listings(data, sfw)
+
+    data = __query_api(api_client, sfw.community.identifier, "nsfw")
+
+    assert len(data["packages"]) == 2
+
+
+@pytest.mark.django_db
+def test_packages_are_filtered_by_required_categories(api_client: APIClient) -> None:
+    site = CommunitySiteFactory()
+    cat1 = PackageCategory.objects.create(
+        name="c1", slug="c1", community=site.community
+    )
+    cat2 = PackageCategory.objects.create(
+        name="c2", slug="c2", community=site.community
+    )
+    PackageListingFactory(community_=site.community)
+    pl2 = PackageListingFactory(community_=site.community, categories=[cat1])
+    pl3 = PackageListingFactory(community_=site.community, categories=[cat2])
+
+    data = __query_api(
+        api_client, site.community.identifier, f"included_categories={cat1.slug}"
+    )
+
+    __assert_packages_by_listings(data, pl2)
+
+    data = __query_api(
+        api_client,
+        site.community.identifier,
+        f"included_categories={cat1.slug}&included_categories={cat2.slug}",
```

Comment:
It dawned on me that this is indeed how Django does list query args by default. We might want to use comma separated slugs with a single key instead, but I'm leaving that under the same "acceptable in experimental" reasoning as with the other query parameter related feedback

Reply:
Any specific reasons why comma-separation would be better?

Reply:
It's just nicer to work with on the clientside quite often (anecdotal evidence), since this form of query parameter is more complex. I'd prefer a single key per field, and comma separated list seems the cleanest way to achieve that, though I'm open to other syntax suggestions.

Thinking about the practical options, it's probably easiest to use list field in conjunction with a `ModelChoiceField`, saving a lot of parsing on the view itself. You should be able to use the same pattern as the `PackageUploadMetadataSerializer` here:

```python
    categories = serializers.ListField(
        child=CommunityFilteredModelChoiceField(
            queryset=PackageCategory.objects.all(),
            to_field="slug",
        ),
        allow_empty=True,
    )
 ```

And then encode them to the URL as `categories=[slug1, slug2, slug3]`

Reply:
Also now that I think about it, using a model choice field might be counter-intuitive for query usage, as we don't necessarily need the category objects, just the slugs. In any case I feel like that should illustrate the design options available quite well.

Reply:
The query parameter parsing was changed, but the test still uses the multi-key format. Is this intentional?

Reply:
Well, there was no intent one way or another behind leaving this as it was. However, the changes affected how e.g. boolean fields are parsed from text to proper types, not how lists are submitted. The deserializer still eats a `QueryDict`, which expects lists to be passed in this way. Otherwise we would have to manually split the received string by the separator character.

Reply:
We could delegate this to a separate issue since it's less about implementation detail and more about API design. Let's not block this PR with it at the very least.

Reply:
Created TS-258 tracking it